### PR TITLE
ddl: disallow dropping index on auto_increment column (#11360)

### PR DESCRIPTION
**Pull request by tangenta**
Cherry-pick #11360.

### CODE REVIEW

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -295,7 +296,7 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
 		}
 	}
 	if (autoIncrementMustBeKey && !isKey) || count > 1 {
-		p.err = errors.New("Incorrect table definition; there can be only one auto column and it must be defined as a key")
+		p.err = autoid.ErrWrongAutoKey.GenWithStackByArgs()
```
Comment (zz-jason) [inline planner/core/preprocess.go]: If arguments is empty, I think we can use `GenWithStack()`

Comment (tangenta) [reply planner/core/preprocess.go]: @zz-jason `GenWithStack()` accepts a formatted string, but there is no such string here.

**Comment by tangenta** [comment]
```
[2019-07-24T11:07:45.144Z] ----------------------------------------------------------------------
[2019-07-24T11:07:45.144Z] FAIL: client_fail_test.go:32: testClientSuite.TestPanicInRecvLoop
[2019-07-24T11:07:45.144Z] 
[2019-07-24T11:07:45.144Z] client_fail_test.go:57:
[2019-07-24T11:07:45.144Z]     c.Assert(err, IsNil)
[2019-07-24T11:07:45.144Z] ... value *errors.fundamental = injected error in batchRecvLoop ("injected error in batchRecvLoop")
```
/run-check_dev_2

**Comment by tangenta** [comment]
```
----------------------------------------------------------------------
FAIL: client_fail_test.go:32: testClientSuite.TestPanicInRecvLoop

client_fail_test.go:57:
    c.Assert(err, IsNil)
... value *errors.fundamental = injected error in batchRecvLoop ("injected error in batchRecvLoop")
```

**Pull request merged**